cpp/src/arrow/compute/kernels/scalar_nested_test.cc

```diff
@@ -39,6 +39,46 @@ TEST(TestScalarNested, ListValueLength) {
   }
 }
 
+TEST(TestScalarNested, ListElement) {
+  for (auto ty : {list(int16()), fixed_size_list(uint8(), 2)}) {
+    auto input = ArrayFromJSON(ty, "[[8, 5], [74, 92], [7, 4]]");
+    auto out_ty = ty->id() == Type::LIST ? int16() : uint8();
+    auto expected = ArrayFromJSON(out_ty, "[5, 92, 4]");
+    auto index = ScalarFromJSON(int32(), "1");
+    CheckVectorBinary("list_element", input, Datum(index), expected);
```

Comment:
We don't need CheckVectorBinary anymore since this is a scalar kernel.

Reply:
What should I use instead?

Reply:
Use CheckScalar. As noted below, you'll need to add support for all-scalar inputs. The advantage is that CheckScalar automatically also tests chunked array and scalar inputs for you.